config/telemetry.go

```diff
@@ -0,0 +1,405 @@
+	// metrics will be sent to that instance.
+	//
+	// hcl: telemetry { statsd_address = string }
+	StatsdAddr string `json:"statsd_address,omitempty" mapstructure:"statsd_address"`
+
+	// StatsiteAddr is the address of a statsite instance. If provided,
+	// metrics will be streamed to that instance.
+	//
+	// hcl: telemetry { statsite_address = string }
+	StatsiteAddr string `json:"statsite_address,omitempty" mapstructure:"statsite_address"`
+
+	// PrometheusRetentionTime is the time before a prometheus metric expires.
+	//
+	// hcl: telemetry { prometheus_retention_time = "duration" }
+	PrometheusRetentionTime time.Duration `json:"prometheus_retention_time,omitempty" mapstructure:"prometheus_retention_time"`
+
+	// PrometheusPort is the REST port under which the metrics can be queried.
+	//
+	// hcl: telemetry { prometheus_port = int }
+	PrometheusPort int `json:"prometheus_port,omitempty" mapstructure:"prometheus_port"`
+}
+
+func DefaultTelemetryConfig() *TelemetryConfig {
+	return &TelemetryConfig{}
+}
+
+func (c *TelemetryConfig) Copy() *TelemetryConfig {
+	if c == nil {
+		return nil
+	}
+
+	return &TelemetryConfig{
+		Disable:                            c.Disable,
+		CirconusAPIApp:                     c.CirconusAPIApp,
+		CirconusAPIToken:                   c.CirconusAPIToken,
+		CirconusAPIURL:                     c.CirconusAPIURL,
+		CirconusBrokerID:                   c.CirconusBrokerID,
+		CirconusBrokerSelectTag:            c.CirconusBrokerSelectTag,
+		CirconusCheckDisplayName:           c.CirconusCheckDisplayName,
+		CirconusCheckForceMetricActivation: c.CirconusCheckForceMetricActivation,
+		CirconusCheckID:                    c.CirconusCheckID,
+		CirconusCheckInstanceID:            c.CirconusCheckInstanceID,
+		CirconusCheckSearchTag:             c.CirconusCheckSearchTag,
+		CirconusCheckTags:                  c.CirconusCheckTags,
+		CirconusSubmissionInterval:         c.CirconusSubmissionInterval,
+		CirconusSubmissionURL:              c.CirconusSubmissionURL,
+		DisableHostname:                    c.DisableHostname,
+		DogstatsdAddr:                      c.DogstatsdAddr,
+		DogstatsdTags:                      c.DogstatsdTags,
+		FilterDefault:                      c.FilterDefault,
+		AllowedPrefixes:                    c.AllowedPrefixes,
+		BlockedPrefixes:                    c.BlockedPrefixes,
+		MetricsPrefix:                      c.MetricsPrefix,
+		StatsdAddr:                         c.StatsdAddr,
+		StatsiteAddr:                       c.StatsiteAddr,
+		PrometheusPort:                     c.PrometheusPort,
+		PrometheusRetentionTime:            c.PrometheusRetentionTime,
+	}
+}
+
+// Merge combines all values in this configuration with the values in the other
+// configuration, with values in the other configuration taking precedence.
+// Maps and slices are merged, most other values are overwritten.
+func (c *TelemetryConfig) Merge(o *TelemetryConfig) *TelemetryConfig {
```

Comment:
can we add tests for this merge function?

Reply:
Sure!
I added some tests as requested